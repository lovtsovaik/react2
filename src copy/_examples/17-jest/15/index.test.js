const API_URL = 'https://jsonplaceholder.typicode.com/todos';

const implementation = (url /*, options */) => {
    return Promise.resolve({
        json() {
            return Promise.resolve(`A call to ${url} was successful`);
        },
    });
};

global.fetch = jest.fn(implementation);

class Api {
    async getTodos() {
        const response = await fetch(API_URL);
        const result = await response.json();

        return result;
    }
}

const api = new Api();

describe('«api.getUsers»:', () => {
    test('should send a fetch request and yield data', async () => {
        await expect(api.getTodos()).resolves.toMatchSnapshot();
    });

    test('should call «fetch» function one time', () => {
        expect(fetch).toHaveBeenCalledTimes(1);
        expect(fetch).not.toHaveBeenCalledTimes(2);
    });

    test('should send a fetch request to valid url', () => {
        expect(fetch).toHaveBeenCalledWith(API_URL);
    });
});
