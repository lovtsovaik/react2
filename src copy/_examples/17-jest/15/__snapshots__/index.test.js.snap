// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`«api.getUsers»: should send a fetch request and yield data 1`] = `"A call to https://jsonplaceholder.typicode.com/todos was successful"`;
